jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -506,6 +507,27 @@ public String toString()
         }
     }
 
+    /**
+     * <p>An unsafe {@link HttpURI} that accepts URI parts without checking whether they are valid.</p>
```

Comment:
```suggestion
     * <p>An unsafe {@link HttpURI} that accepts URI parts without checking whether they are valid.
     * Primarily intended for testing, as un-validated may break arbitrary code.</p>
```

Reply:
It's also used by Jetty HttpClient to allow it to make requests that other servers would allow, but we don't.

An example would be using the `%00` in the request path.
In jetty-12.0.x HEAD our HttpURI and MetaData classes both reject that pct-encoding, but it's 100% allowed to be sent from a user-agent.